### ISSUE TITLE
RunYesNoDialog Fails

### ISSUE THREAD

**Issue by tbpassin** [leo-editor/leo-editor#4012]
See PR #4020.

[Later] This has turned out to be caused by the `RunYesNoDialog()` class failing to return the correct values.]

If you edit a node that is not an external file with an external editor and save it in that editor, Leo detects that the outline has changed and puts up the right dialog about reloading it, but does not actually reload it. This could happen if a second Leo window has been opened and the node edited in it, then saved.  It could also happen if the node has been opened for editing by a script or menu item having a body like this:
```
ext:
kind:   subprocess.Popen
arg:    "C:\Program Files\EditPlus\editplus.exe"
```
To reproduce:
Open Leo and create a new node in the workbook.  Open a second Leo session, navigate to the same node, and add some text.  Save the outline.  Click on the first Leo window.  Notice that the _File Changed, Reload It?_ dialog opens.  Click on "yes".  Observe that the node has not changed.

This functionality used to work.  I don't know when it stopped working but I noticed it today.
```
Leo 6.8.0, 6.8.0 branch, build 4599d84609
2024-06-25 03:38:07 -0500
Python 3.12.3, PyQt version 6.7.2
Windows 10 AMD64 (build 10.0.19045) SP0
```

**Comment by tbpassin** [comment]
The `revert` command isn't reverting the outline to the one on disk.  This is probably the reason.  To test:

1. Save outline; 
2. Type a new line of text;
3. Without saving, issue the `revert` minibuffer command.

Actual result: the added line remains.
Expected result: the added line disappears.

**Comment by tbpassin** [comment]
This is because in _c_file.revert()_ the dialog for reverting is returning "cancel" instead of "yes" when the "Yes" button is clicked:

```
    reply = g.app.gui.runAskYesNoDialog(
        c, 'Revert', f"Revert to previous version of {fn}?")

```

**Comment by tbpassin** [comment]
Getting closer: the dialog is returning `2` but the code expects a value of `0`.

**Comment by tbpassin** [comment]
A test script:
```
gui = g.app.gui

title = 'Test Dialog'
msg = 'Try Something'
result = gui.runAskYesNoDialog(c, title, msg)
g.es(result)
```

**Comment by tbpassin** [comment]
The two dialog buttons return 2 and 3, not 1 and 2.

Presumably the optional buttons `yes_all` and `no_all` also return unexpected values but I haven't checked that.

This could be an issue with other dialogs.

**Comment by tbpassin** [comment]
Found it. The buttons return their ButtonRole values when clicked. The _Yes_ button has been given the `ButtonRole.Yes`  role, and that enum role is `5`, not `0`.

It looks like the code assumes that the return value will be the order in which the buttons were defined.  Maybe it was that way once, but not now.

ref: https://doc.qt.io/qt-6/qmessagebox.html#ButtonRole-enum

**Comment by edreamleo** [comment]
Fixed via PR #4012.
